searx/poolrequests.py

```diff
@@ -87,15 +87,38 @@ def get_time_for_thread():
     return threadLocal.total_time
 
 
+def get_proxy_cycles():
+    proxy_settings = settings['outgoing'].get('proxies')
+    if proxy_settings:
```

Comment:
if `outgoing.proxies` is not defined in settings.yml, the result of this function is `None`, on line 108 it will trigger `'NoneType' object is not iterable`.

```python
proxy_settings = settings['outgoing'].get('proxies')
if not proxy_settings:
   return None
...
```

later in get_proxies
```python
def get_proxies():
    global proxy_cycles
    if not proxy_cycles:
        return None
    ...
```